schema/encoding/jsonschema/benchmark_complex_test.go

```diff
@@ -0,0 +1,221 @@
+package jsonschema_test
+
+import (
+	"bytes"
+	"github.com/rs/rest-layer/schema"
+	"github.com/rs/rest-layer/schema/encoding/jsonschema"
+	"math"
+	"testing"
+	"time"
+)
+
+// Optional schema field
+func Optional(s schema.Field) schema.Field {
+	s.Required = false
+	return s
+}
+
+// Default modifies an existing schema.Field with a default value.
+func Default(f schema.Field, v interface{}) schema.Field {
+	f.Default = v
+	return f
+}
+
+// Required modifies an existing schema.Field to be required.
+func Required(s schema.Field) schema.Field {
+	s.Required = true
+	return s
+}
+
+func rfc3339Nano() schema.Field {
+	return schema.Field{
+		Required:    false,
+		Description: "UTC start time in RFC3339 format with Nano second support, e.g. 2006-01-02T15:04:05.999999999Z",
+		Validator: &schema.Time{
+			TimeLayouts: []string{time.RFC3339Nano},
+		},
+	}
+}
+
+// String field
+func String(min, max int, description string) schema.Field {
+	return schema.Field{
+		Description: description,
+		Required:    true,
+		Default:     "",
+		Validator: &schema.String{
+			MinLen: min,
+			MaxLen: max,
+		},
+	}
+}
+
+// Integer returns schema.Field template for int validation.
+func Integer(min, max int, description string) schema.Field {
+	return schema.Field{
+		Default:     min,
+		Description: description,
+		Validator: &schema.Integer{
+			Boundaries: &schema.Boundaries{
+				Min: float64(min),
+				Max: float64(max),
+			},
+		},
+	}
+}
+
+// Bool validator helper
+func Bool(description string) schema.Field {
+	return schema.Field{
+		Description: description,
+		Required:    true,
+		Validator:   &schema.Bool{},
+	}
+}
+
+var (
```

Comment:
All this schemas are left in the namespace... Perhaps it would be better just to leave a single ComplexSchema/complexSchema and initalize it in an `init()` function.